### PR TITLE
Creates SAML Okta configuration guide

### PR DESCRIPTION
**Pull request by bstuder99**
## Proposed Changes

Ports [this doc](https://github.com/kubecost/poc-common-configurations/tree/main/saml-okta) from the poc-common-config repo, updates some of the language/instructions used to be more clear. Restructures/rewords our SAML/OICD content to be more descriptive as well

### CODE REVIEW

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+```
+saml:
+   encryptionCertSecret: "kubecost-saml-cert"
+   decryptionKeySecret: "kubecost-saml-decryption-key"
+```
+
+## Troubleshooting
+
+You can view the logs on the cost-model container. In this example, the assumption is that the prefix for Kubecost groups is `kubecost_`. This script is currently a work in progress.
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:184]: Nit

```suggestion
You can view the logs on the cost-model container. In this example, the assumption is that the prefix for Kubecost groups is `kubecost_`. This command is currently a work in progress.

```

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+`kubectl create secret generic kubecost-okta --from-file myservice.cert --namespace kubecost`
+
+{% hint style="info" %}
+For configuring single app logout, read [Okta's documentation](https://help.okta.com/en-us/content/topics/apps/apps_single_logout.htm) on the subject. then, update the `values.saml:redirectURL`value in your [_values.yaml_](https://github.com/kubecost/poc-common-configurations/blob/main/saml-okta/values-saml.yaml) file.
+{% endhint %}
+
+10. Use this [Okta document](https://help.okta.com/en-us/content/topics/apps/apps-manage-assignments.htm) to assign individuals or groups access to your Kubecost application.
+11. Finally, add `-f values-saml.yaml` to your Kubecost Helm upgrade command:
+
+```
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:38]: Should have language set for code block
```suggestion
```shell
```

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+At this point, test your SSO to ensure it is working properly before moving on to the next section.
+
+## Okta RBAC configuration (admin/readonly)
+
+The simplest form of RBAC in Kubecost is to have two groups: `admin` and `readonly`. If your goal is to simply have these two groups, you do not need to configure filters. This will result in the logs message: `file corruption: '%!s(MISSING)'`, but this is expected.
+
+The [values-saml.yaml](https://github.com/kubecost/poc-common-configurations/blob/main/saml-okta/values-saml.yaml) file contains the `admin` and `readonly` groups in the RBAC section:
+
+```
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:53]: Should have language set for code block
```suggestion
```yaml
```

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+Filters are used to give visibility to a subset of objects in Kubecost. Examples of the various filters available are in [filters.json](https://github.com/kubecost/poc-common-configurations/blob/main/saml-okta/filters.json) and [filters-examples.json](https://github.com/kubecost/poc-common-configurations/blob/main/saml-okta/filters-examples.json). RBAC filtering is capable of all the same types of filtering features as that of the [Allocation API](/apis/apis-overview/api-allocation.md).
+
+It's possible to combine filtering with admin/readonly rights
+
+These filters can be configured using groups or user attributes in your Okta directory. It is also possible to assign filters to specific users. The example below is using groups.
+
+Filtering is configured very similarly to the admin/readonly above. The same group pattern match (kubecost_group) can be used for both, as is the case in this example:
+
+```
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:82]: Should have language set for code block
```suggestion
```yaml
```

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    customGroups: # not needed for simple admin/readonly RBAC
+      - assertionName: "kubecost_group"
+```
+
+The array of groups obtained during the authorization request will be matched to the subject key in the *filters.json*:
+
+```
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:89]: Should have language set for code block
```suggestion
```json
```

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+3. Select each group, then select *Assign people* and add the appropriate users for testing. Select _Done_ to confirm edits to a group. Kubecost admins will be part of both the read only *kubecost_users* and *kubecost_admin groups*. Kubecost will assign the most rights if there are conflicts.
+
+4. Return to the Groups page. Select *kubecost_users*, then in the _Applications_ tab, assign the Kubecost application. You do not need to assign the other *kubecost_* groups to the Kubecost application because all users already have access in the *kubecost_users* group.
+
+5. Modify filters.json as depicted above.
+
+6. Create the ConfigMap using the following command:
+
+```
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:136]: Should have language set for code block
```suggestion
```shell
```

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+kubectl create configmap group-filters --from-file filters.json -n kubecost
+```
+
+You can modify the ConfigMap without restarting any pods.
+
+```
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:142]: Should have language set for code block
```suggestion
```shell
```

File: install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+
+`kubectl create secret generic kubecost-saml-cert --from-file saml-encryption-cert.cer --namespace kubecost`
+
+6. Create a secret with the private key. The file name **must** be *saml-encryption-key.pem*.
+
+`kubectl create secret generic kubecost-saml-decryption-key --from-file saml-encryption-key.pem --namespace kubecost`
+
+7. Pass the following values via Helm into your *values.yaml*:
+
+```
```
Comment (srpomeroy) [inline install-and-configure/advanced-configuration/user-management-saml/okta-saml-integration.md:176]: Should have language set for code block
```suggestion
```yaml
```

**Review by srpomeroy — CHANGES_REQUESTED** [review]
Some minor stuff.

We use a single backtick and fenced code blocks to represent commands. In this doc specifically, both are used interchangeably for similar commands. Do we have a pattern we like to follow?

**Comment by bstuder99** [comment]
I would say I use triple backticks for code blocks, with single backticks when referring to a single line or quoting smaller-scale content from the CLI

**Comment by bstuder99** [comment]
This is the difference in Markdown from inline text and code blocks

**Pull request merged**